gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,261 @@
+        //
+        COL_TYPE,
+        TYPE_CONTENT);
+  }
+
+  /**
+   * Get the last written run id which is having a marker row.
+   *
+   * @return latest completed run id.
+   */
+  public String getLatestCompletedRunID() {
+    // collect marker row for the last written run id
+    // Example query:
+    // SELECT gcRunId FROM nessie.db1.`identified_results@someGcRef
+    //    WHERE type = 'GC_MARK' ORDER BY gcRunStart DESC LIMIT 1
+    List<Row> markerRows =
+        sql(
+                "SELECT %s FROM %s WHERE %s = '%s' ORDER BY %s DESC LIMIT 1",
+                COL_GC_RUN_ID,
+                //
+                catalogAndTableWithRefName,
+                //
+                COL_TYPE,
+                TYPE_GC_MARKER,
+                //
+                COL_GC_RUN_START)
+            .collectAsList();
+    return markerRows.isEmpty() ? null : markerRows.get(0).getString(0);
+  }
+
+  /**
+   * Convert each task result into a spark row and store it in the output table. Write a marker row
+   * in the end to indicate that the task is completed.
+   *
+   * @param rowDataset content rows to be written.
+   * @param runId run id of current identify task.
+   * @param startedAt task start time.
+   */
+  void writeToOutputTable(Dataset<Row> rowDataset, String runId, Timestamp startedAt) {
+    try {
+      // write content rows
+      rowDataset.writeTo(catalogAndTableWithRefName).append();
+      // write marker row
+      sparkSession
+          .createDataFrame(Collections.singletonList(createMarkerRow(runId, startedAt)), schema)
+          .writeTo(catalogAndTableWithRefName)
+          .append();
+    } catch (NoSuchTableException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static Row getContentRowVariablePart(
+      String contentId, String contentType, long snapshotId, String refName, String hash) {
+    return RowFactory.create(
+        IdentifiedResultsRepo.TYPE_CONTENT,
+        // the fixed value columns (runId and startTime)
+        // will be added in the callers using dataframe.withColumn().
+        // Hence they are filled as 'null' here.
+        null,
+        null,
+        contentId,
+        contentType,
+        snapshotId,
+        refName,
+        hash);
+  }
+
+  private static Row createMarkerRow(String runID, Timestamp startedAt) {
+    return RowFactory.create(
+        IdentifiedResultsRepo.TYPE_GC_MARKER, startedAt, runID, null, null, null, null, null);
+  }
+
+  private void createTableIfAbsent(
+      SparkSession sparkSession,
+      String catalogName,
+      TableIdentifier tableIdentifier,
+      String gcRefName) {
+    // get Nessie catalog
+    Catalog nessieCatalog =
+        CatalogUtil.loadCatalog(
+            NessieCatalog.class.getName(),
+            catalogName,
+            catalogConfWithRef(sparkSession, catalogName, gcRefName),
+            sparkSession.sparkContext().hadoopConfiguration());
+    try {
+      nessieCatalog.createTable(tableIdentifier, icebergSchema);
```

Comment:
How do you prevent that GC doesn't expire it's own data?

Reply:
I think based on previous internal discussion.
GC output table should be considered as another normal iceberg table.
So it should be OK to GC its own contents and expire them. When I add a documentation on GC, I will update this part.

Reply:
Documentation isn't enough IMO.
There will be users that will shoot themselves into the foot.

Reply:
GC output table is just another iceberg table created in separate reference.

> The fact that GC can wipe its own data

I am not sure about above statement, 

a) If the gc output table is not dropped,
gc don't delete the actual gc output rows/data, only redundant expired metadata (like snapshot id form metadata) will be removed.
b) If the gc output table is dropped, as it is another Nessie managed Iceberg table, actual data of dropped gc output table will be cleaned from GC.

Are you suggesting don't keep it as Nessie managed table? or skip gc for this table?

Reply:
> or skip gc for this table?
How many dates the GC will write? Wouldn't it lead to too much data being kept? Maybe we should have a similar `cutoffPeriod` for the GC table as well? (but it would be substantially higher)